pkg/engine/armresources_test.go

```diff
@@ -1191,3 +1191,662 @@ func TestGenerateARMResourceWithVMASAgents(t *testing.T) {
+	var cs api.ContainerService
+
+	json.Unmarshal([]byte(apiModelStr), &cs)
+
+	armResources := GenerateARMResources(&cs)
+
+	tg, _ := InitializeTemplateGenerator(Context{})
+
+	expectedCustomDataStr := getCustomDataFromJSON(tg.GetKubernetesLinuxNodeCustomDataJSONObject(&cs, cs.Properties.AgentPoolProfiles[0]))
+
+	var dataDisks []compute.VirtualMachineScaleSetDataDisk
+
+	var userAssignedIDEnabled = cs.Properties.OrchestratorProfile.KubernetesConfig.UserAssignedIDEnabled()
+
+	agentLoadBalancer := LoadBalancerARM{
+		ARMResource: ARMResource{
+			APIVersion: "[variables('apiVersionNetwork')]",
+			DependsOn: []string{
+				"[concat('Microsoft.Network/publicIPAddresses/', variables('agentPublicIPAddressName'))]",
+			},
+		},
+		LoadBalancer: network.LoadBalancer{
+			Location: to.StringPtr("[variables('location')]"),
+			Name:     to.StringPtr("[variables('agentLbName')]"),
+			LoadBalancerPropertiesFormat: &network.LoadBalancerPropertiesFormat{
+				BackendAddressPools: &[]network.BackendAddressPool{
+					{
+						Name: to.StringPtr("[variables('agentLbBackendPoolName')]"),
+					},
+				},
+				FrontendIPConfigurations: &[]network.FrontendIPConfiguration{
+					{
+						Name: to.StringPtr("[variables('agentLbIPConfigName')]"),
+						FrontendIPConfigurationPropertiesFormat: &network.FrontendIPConfigurationPropertiesFormat{
+							PublicIPAddress: &network.PublicIPAddress{
+								ID: to.StringPtr("[resourceId('Microsoft.Network/publicIpAddresses',variables('agentPublicIPAddressName'))]"),
+							},
+						},
+					},
+				},
+				OutboundRules: &[]network.OutboundRule{
+					{
+						OutboundRulePropertiesFormat: &network.OutboundRulePropertiesFormat{
+							FrontendIPConfigurations: &[]network.SubResource{
+								{
+									ID: to.StringPtr("[variables('agentLbIPConfigID')]"),
+								},
+							},
+							BackendAddressPool: &network.SubResource{
+								ID: to.StringPtr("[concat(variables('agentLbID'), '/backendAddressPools/', variables('agentLbBackendPoolName'))]"),
+							},
+							Protocol:             network.Protocol1All,
+							IdleTimeoutInMinutes: to.Int32Ptr(15),
```

Comment:
how did you pick this value? :)

Reply:
Got this value from the samples showed in the Azure loadbalancer docs. https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-tcp-idle-timeout Looks like we currently have a mix of `5` and `30` from master. Do you have a suggestion?

Reply:
this number is very close to the default Linux idle timeout (15.4 min): https://pracucci.com/linux-tcp-rto-min-max-and-tcp-retries2.html

I would think we set value higher than that number, so the LB don't give up before OS does.